### PR TITLE
Tracking branch for a large bundle of updates.

### LINKED ISSUE #46: Link in the initial create process leads to a 404
During the create process the initial screen displays a link which leads to a 404, this one https://open-wc.org/init/
I'm currently busy to fix a typo anyways, so I could change it, but to where should the link point?

Should it link to here: https://open-wc.org/docs/development/generator/ ?

I also see that it (the initial screen with the link) is mentioned in the open-wc repository. Is there a link between them or do they need to be updated separately?

### LINKED ISSUE #79: Error while transforming node_modules/@web/storybook-prebuilt/manager.js: Could not resolve import "react".
## Actual Behavior
I used following command to create an application skeleton:
```shell
npm init @open-wc --destinationPath /Users/lukaskoller/Desktop/ui-demo --type scaffold --scaffoldType wc --features linting testing demoing --typescript true --tagName ui-demo --writeToDisk true --installDependencies npm 
```
Starting storybook using `npm run storybook` leads to:
```shell
Error while transforming node_modules/@web/storybook-prebuilt/manager.js: Could not resolve import "react".
```

## Expected Behavior
storybook dev server starts without errors and I can inspect storybook on `http://localhost:8000`.

### LINKED ISSUE #87: Storybook Does Not Work Out of the Box
After initializing a new component with the following options:
- Scaffold a new project
- Web Component
- Linting (eslint & prettier)
- Testing (web-test-runner)
- Demoing (storybook)
- Typescript
- npm

and running `npm run storybook`, storybook opens, but shows an error saying:
```shell

Couldn't find any stories in your Storybook.

- Please check your stories field of your main.js config.
- Also check the browser console and terminal for error messages.
```

It's unclear to me what is causing the error, as the glob in `.storybook/main.js` appears to be correct, and the `<component-name>.stories.js` files are built in dist.

### LINKED ISSUE #89: [app] Storybook Does Not Work Out of the Box
After initializing a new component with the following options:

Scaffold a new project
Application
Linting (eslint & prettier)
Testing (web-test-runner)
Demoing (storybook)
Typescript
npm
and running npm run storybook, the following error is thrown:

```shell
interneuron:st-app kito$ npm run storybook

> st-app@0.0.0 storybook
> tsc && npm run analyze -- --exclude dist && concurrently -k -r "tsc --watch --preserveWatchOutput" "wds -c .storybook/server.mjs"


> st-app@0.0.0 analyze
> cem analyze --litelement --exclude dist

[11:38:31 PM] @custom-elements-manifest/analyzer: Created new manifest.
[11:38:31 PM] Starting compilation in watch mode...

Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/kito/Core/Projects/focus.speakertrax/vcs/st-speaker-profile/st-app/.storybook/main.js from /Users/kito/Core/Projects/focus.speakertrax/vcs/st-speaker-profile/st-app/node_modules/@web/dev-server-storybook/dist/shared/config/readStorybookConfig.js not supported.
main.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead rename main.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in /Users/kito/Core/Projects/focus.speakertrax/vcs/st-speaker-profile/st-app/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).

    at Object.readStorybookConfig (/Users/kito/Core/Projects/focus.speakertrax/vcs/st-speaker-profile/st-app/node_modules/@web/dev-server-storybook/dist/shared/config/readStorybookConfig.js:56:35)
    at storybookPlugin (/Users/kito/Core/Projects/focus.speakertrax/vcs/st-speaker-profile/st-app/node_modules/@web/dev-server-storybook/dist/serve/storybookPlugin.js:16:51)
    at file:///Users/kito/Core/Projects/focus.speakertrax/vcs/st-speaker-profile/st-app/.storybook/server.mjs:7:13 {
  code: 'ERR_REQUIRE_ESM'
}
```
This is potentially related to #87.

### PR DESCRIPTION
**Pull request by Westbrook**
After running the CLI, I found that a number of the dependencies were out of date, or used deprecated configs, etc. I manually updated a generated project and have been back porting the required change to the CLI piece by piece.

## What I did

1. Made a bunch of stand alone updates in individual PRs to this branch then overtime will hopefully catch the whole generator up to "modern" 😜 dependencies and practices.
  - [x] #94 (bigger change) 
  - [x] #95
  - [x] #96 
  - [x] #97 
  - [x] #98 (bigger change)
  - [x] #99 
  - [x] #100 
  - [x] #101 
  - [x] #102
  - [x] #103 
  - [x] #105  
  - [x] #106  
  - [x] #107  
  - [x] #108  
  - [x] #109  
  - [x] #110
  - [x] #111
  - [x] #113
  - [x] #116
  - [x] #117
  - [x] #118
  - [x] #119 
  - [x] #120
  - [x] #121
  - [x] #122
  - [x] #126
  - [x] Some more PRs coming soon?

I'd love to include an update to `eslint@9.x` but https://github.com/import-js/eslint-plugin-import doesn't seem ready to make the update... so, I guess skipping for now is OK.

CI doesn't test the CLI in a bullet proof way, but I'll be doing some more manual testing after all of the individual PRs have merged.

fixes #89
fixes #87
fixes #79
fixes #46

### CODE REVIEW

**Comment by Westbrook** [comment]
This now works as expected for projects generated with:
- Scaffold
- App
- lint, test, demo, build
- ts, yes
- write to disc
- npm

I'm gonna step through _some_ of the other options over the next day or two, but this may be ready to go to `master` as is, in that many things don't work there anyways and a full overview of all of the things is difficult to achieve by a single tester.

Also, `renovate` running free into `master` makes maintaining a long living branch (even just a few days) a bit of a pita.

**Comment by Westbrook** [comment]
The following options also work as expected:
- Scaffold
- WC
- lint, test, demo
- ts, no
- npm

🥳

File: src/generators/building-rollup/templates/package.json [hunk 1]
```diff
@@ -4,17 +4,16 @@
     "start:build": "web-dev-server --root-dir dist --app-index index.html --open"
   },
   "devDependencies": {
-    "@babel/preset-env": "^7.16.11",
-    "@open-wc/building-rollup": "^2.0.2",
-    "@rollup/plugin-babel": "^5.3.1",
-    "@rollup/plugin-node-resolve": "^13.0.6",
-    "@web/rollup-plugin-html": "^1.11.1",
-    "@web/rollup-plugin-import-meta-assets": "^1.0.8",
+    "@babel/preset-env": "^7.24.6",
```
Comment (thepassle) [inline src/generators/building-rollup/templates/package.json:7]: do we even need babel/preset-env anymore? Don't we use the esbuild plugin for this?

Comment (thepassle) [reply src/generators/building-rollup/templates/package.json:7]: It looks like its not used in the rollup config at all anymore, only babel-plugin-template-html-minifier is used, so I guess we can remove it

**Comment by thepassle** [comment]
> I'd love to include an update to eslint@9.x but https://github.com/import-js/eslint-plugin-import doesn't seem ready to make the update... so, I guess skipping for now is OK.

Maybe we should just switch to https://github.com/antfu/eslint-plugin-import-x?

File: src/generators/building-rollup/templates/package.json (hunk 1 above)
Comment (Westbrook) [reply src/generators/building-rollup/templates/package.json:7]: Nice call!! I'll get on that.

**Comment by peschee** [comment]
We'll have to update the `README.md` as well here. This still has things like

>npm init requires node 14 & npm 6 or higher

Does this updated version still work with these Node / NPM versions? I'd assume not.

**Comment by peschee** [comment]
Do you think we could Switch to SB 8.x? This branch still uses `7.6.19`.

**Comment by thepassle** [comment]
> Do you think we could Switch to SB 8.x? This branch still uses 7.6.19.

The storybook builder doesnt support v8 yet

**Comment by Westbrook** [comment]
> Do you think we could Switch to SB 8.x? This branch still uses `7.6.19`.

This is the work going into that: https://github.com/modernweb-dev/web/pull/2674 If you were able to lend some time to supporting the process, we'd definitely get there, sooner. I may have some time to put into that, soonish, but I have to focus on one upgrade at a time...

**Comment by peschee** [comment]
🎉

**Pull request merged**